simbad2k/simbad2k.py

```diff
@@ -96,67 +103,120 @@ def __init__(self, query, scheme):
         self.scheme = scheme
 
     def _clean_result(self, result):
+        """
+        Converts the results from the MPC into a dictionary of floats.
+        Extracts the object's name from the query results and adds it to the dictionary.
+        """
         cleaned_result = {}
         for key in self.keys:
             try:
                 value = float(result[key])
             except (ValueError, TypeError):
                 value = None
             cleaned_result[key] = value
+        # Build object name from returned Data
+        if result['number']:
```

Comment:
Do we want to do a safe `result.get('number')` key check here to make sure we don't get a KeyError if that key doesn't exist?

Reply:
There shouldn't never be a number. Name isn't given for comets.
However, best not to crash if things change...